localstack/services/s3/utils.py

```diff
@@ -140,19 +141,41 @@ def get_bucket_from_moto(
     try:
         return moto_backend.get_bucket(bucket_name=bucket)
     except MissingBucket:
-        ex = NoSuchBucket("The specified bucket does not exist")
-        ex.BucketName = bucket
-        raise ex
+        raise NoSuchBucket("The specified bucket does not exist", BucketName=bucket)
 
 
 def get_key_from_moto_bucket(
-    moto_bucket: moto_s3_models.FakeBucket, key: ObjectKey
-) -> moto_s3_models.FakeKey:
-    fake_key = moto_bucket.keys.get(key)
+    moto_bucket: FakeBucket,
+    key: ObjectKey,
+    version_id: str = None,
+    raise_if_delete_marker_method: Literal["GET", "PUT"] = None,
```

Comment:
I really dislike this, but this is a quick fix for now and would need to be revisited to better implement versioning.